### DISCUSSION TITLE
Alow AutoWrap Text for Buton's text

### DISCUSSION THREAD

**Discussion by IllustriousKnives** [godotengine/godot-proposals#6200] [GUI]
It's a annoying problem: 

- When the Text is too long for the UI --> the UI changes its size ( and its Theme StyleBoxTexture too, deforming it )

So, an AutoWrap Text option for Butons should solves this problem by reducing the Text Size to keep inside the UI .

Thanks

**Comment by Calinou** [comment]
If you autowrap text in a button, you will most likely end up with the same problem on the other axis. The button becomes taller, and your UI will likely be messed up as a result. If you really want to achieve it, you can use a Label node placed within a Button, but this isn't recommended as you'll have to style the Label manually (for hover/pressed/disabled effects).

In most situations, text ellipsis is a better solution to this problem. It's supported in 4.0 as a property in the Button and Label nodes.
